### PR TITLE
[BK-SUPPORT-PART-4] Support bookkeeper autorecovery commands

### PR DESCRIPTION
**Pull request by zymap**
---

Master Issue: #127

*Motivation*

Support bookkeeper autorecovery commands in Pulsarctl.

*Modifications*

Add bookkeeper autorecovery commands.

*Output*

```
➜  pulsarctl-yong git:(autorecovery_cmd) ./pulsarctl bookkeeper autorecovery
Operations about ledger

Usage: pulsarctl bookkeeper autorecovery [flags]

Commands:
  decommission  Decommission a bookie
  getdelay    Get the lost bookie recovery delay of a bookie
  listunderreplicatedledger  List all the underreplicated ledgers which have been marked for rereplication
  recoverbookie  Recover the ledger data of a failed bookie
  setdelay    Set the lost bookie recovery delay
  triggeraudit  Trigger audit by resetting the lostBookieRecoveryDelay
  whoisauditor  Get who is the auditor

Aliases: autorecovery,

Common flags:
  -s, --admin-service-url string    The admin web service url that pulsarctl connects to. (default "http://localhost:8080")
      --auth-params string          Authentication parameters are used to configure the public and private key files required by tls
                                     For example: "tlsCertFile:val1,tlsKeyFile:val2"
      --bookie-service-url string   The bookie web service url that pulsarctl connects to. (default "http://localhost:8080")
  -C, --color string                toggle colorized logs (true,false,fabulous) (default "true")
  -h, --help                        help for this command
      --tls-allow-insecure          Allow TLS insecure connection
      --tls-trust-cert-pat string   Allow TLS trust cert file path
      --token string                Using the token to authentication
      --token-file string           Using the token file to authentication
  -v, --verbose int                 set log level, use 0 to silence, 4 for debugging (default 3)

Use 'pulsarctl bookkeeper autorecovery [command] --help' for more information about a command.
```

### CODE REVIEW

File: pkg/bkctl/autorecovery/autorecovery.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import (
+	"github.com/streamnative/pulsarctl/pkg/cmdutils"
+
+	"github.com/spf13/cobra"
+)
+
+func Command(flagGrouping *cmdutils.FlagGrouping) *cobra.Command {
+	resourceCmd := cmdutils.NewResourceCmd(
+		"autorecovery",
+		"Operations about ledger",
```
Comment (sijie) [inline pkg/bkctl/autorecovery/autorecovery.go]: this command is not about operations of ledgers.

File: pkg/bkctl/autorecovery/decommission.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+package autorecovery
+
+import (
+	"github.com/streamnative/pulsarctl/pkg/cmdutils"
+)
+
+func decommissionCmd(vc *cmdutils.VerbCmd) {
+	var desc cmdutils.LongDescription
+	desc.CommandUsedFor = "This command is used for decommission a bookie."
```
Comment (sijie) [inline pkg/bkctl/autorecovery/decommission.go]:
```suggestion
	desc.CommandUsedFor = "This command is used for decommissioning a bookie."
```

File: pkg/bkctl/autorecovery/get_lost_bookie_recovery_delay.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	var out []cmdutils.Output
+	successOut := cmdutils.Output{
+		Desc: "normal output",
+		Out:  "lostBookieRecoveryDelay value: (delay)",
+	}
+	out = append(out, successOut)
+	desc.CommandOutput = out
+
+	vc.SetDescription(
+		"getdelay",
```
Comment (sijie) [inline pkg/bkctl/autorecovery/get_lost_bookie_recovery_delay.go]: `getdelay` is a very ambiguous verb. Please use "get-lost-bookie-recovery-delay"

File: pkg/bkctl/autorecovery/list_under_replicated_ledger.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+	var include string
+	var exclude string
+	var show bool
+
+	vc.SetRunFunc(func() error {
+		return doListUnderReplicatedLedger(vc, include, exclude, show)
+	})
+
+	vc.FlagSetGroup.InFlagSet("List Under Replicated Ledger", func(set *pflag.FlagSet) {
```
Comment (sijie) [inline pkg/bkctl/autorecovery/list_under_replicated_ledger.go]:
```suggestion
	vc.FlagSetGroup.InFlagSet("List under replicated ledgers", func(set *pflag.FlagSet) {
```

**Comment by zymap** [comment]
0.2.0 will release soon. Let's move this to 0.3.0

**Comment by zymap** [comment]
@sijie PTAL. I will using the testcotainer to add the test in another pull request.

**Pull request merged**